pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -0,0 +1,174 @@
+package csicontrollerset
+
+import (
+	"context"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/staticresourcecontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/library-go/pkg/operator/csi/credentialsrequestcontroller"
+	"github.com/openshift/library-go/pkg/operator/csi/csidrivercontroller"
+)
+
+// CSIDriverControllerOptions contains file names for manifests for both CSI Driver Deployment and DaemonSet.
+type CSIDriverControllerOptions struct {
+	controllerManifest string
+	nodeManifest       string
+}
+
+// CSIDriverControllerOption is a modifier function for CSIDriverControllerOptions.
+type CSIDriverControllerOption func(*CSIDriverControllerOptions)
+
+// WithControllerService returns a CSIDriverControllerOption
+// with a Deployment (CSI Controller Service) manifest file.
+func WithControllerService(file string) CSIDriverControllerOption {
+	return func(o *CSIDriverControllerOptions) {
+		o.controllerManifest = file
+	}
+}
+
+// WithNodeService returns a CSIDriverControllerOption
+// with a DaemonSet (CSI Node Service) manifest file.
+func WithNodeService(file string) CSIDriverControllerOption {
+	return func(o *CSIDriverControllerOptions) {
+		o.nodeManifest = file
+	}
+}
+
+// CSIControllerSet contains a set of controllers that are usually used to deploy CSI Drivers.
+type CSIControllerSet struct {
+	logLevelController           factory.Controller
+	managementStateController    factory.Controller
+	staticResourcesController    factory.Controller
+	credentialsRequestController factory.Controller
+	csiDriverController          *csidrivercontroller.CSIDriverController
+
+	operatorClient v1helpers.OperatorClient
+	eventRecorder  events.Recorder
+}
+
+// Run starts all controllers initialized in the set.
+func (c *CSIControllerSet) Run(ctx context.Context, workers int) {
```

Comment:
at some point, not in the PR, I think we want a `.Complete` step like we use in apiservers that can let users know if they're missing a new controller.  We added a `WithoutX` to one of them.

Reply:
Ack. Added to my TODO